libsupport/src/Experimental.cpp

```diff
@@ -0,0 +1,132 @@
+#include "katana/Experimental.h"
+
+#include <iomanip>
+#include <memory>
+
+#include "katana/Env.h"
+#include "katana/Logging.h"
+#include "katana/Strings.h"
+
+namespace {
+
+class ExperimentalFeatureEnvState {
+public:
+  static ExperimentalFeatureEnvState* Get() {
+    std::call_once(init_flag_, [&]() {
+      state_ = std::unique_ptr<ExperimentalFeatureEnvState>(
+          new ExperimentalFeatureEnvState());
+    });
+    return state_.get();
+  }
+
+  bool WasInEnv(const std::string& feature) {
```

Comment:
The past tense here is throwing me off.  Isn't this `WasUsed`?

Reply:
Here I'm talking about the snapshot that I took of the environment. I thought `IsInEnv` would imply that I am checking the environment every time.

Elsewhere I'm using `Used` to talk about features listed in the environment that map to features that were declared. `WasInEnv` was meant to be the union of `Used` and `Unrecognized`.